viewing_party/party.py

```diff
@@ -1,23 +1,164 @@
+        average = sum / length
+
+    return average   
+
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    length = len(user_data["watched"])
+    
+    if length > 0 :
+        for i in range(length):
+            genre = user_data["watched"][i]["genre"]
+            if genre in genre_dict.keys():
+                genre_dict[genre] +=1
+            else :
+                genre_dict[genre] = 1
+
+        
+        genre_list = []
+        for k, v in genre_dict.items():
+            genre_list.append((v, k))   
+        genre_list.sort(reverse = True)
+        return genre_list[0][1]
+    else :
+        return None
+         
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    user_unique_watched = []
+    friends_watched_list =[]
+    user_list = user_data["watched"]
+    friends= user_data["friends"]
+    
+    for friend in friends:
+        friend_watched = friend["watched"]
+        friends_watched_list += friend_watched
+
+    for movie in user_list:
+        if movie not in friends_watched_list:
+            user_unique_watched.append(movie)   
+
+    return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    friends_watch_list = []
+    user_list = user_data["watched"]
+    friends=user_data["friends"]
+
+    for friend in friends:
+        friend_watched = friend["watched"]
+        friends_watch_list +=friend_watched
+
+    for movie in friends_watch_list:
+        if movie not in user_list and movie not in friends_unique_watched:
+            friends_unique_watched.append(movie)
+
+    return friends_unique_watched           
+
+    
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    friends_unique_movies = get_friends_unique_watched(user_data)
+
+    for movie in friends_unique_movies:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+
+    return recommended_movies        
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+    genre = get_most_watched_genre(user_data)
+    friends_unique_watched = get_friends_unique_watched(user_data)
+
+    for movie in friends_unique_watched :
+        if movie["genre"] == genre :
+            recommended_movies.append(movie)
+
+    return recommended_movies        
+
+def get_rec_from_favorites(user_data):
+    recommended_movies = []
+    friends_all_movies = []
+    friends = user_data["friends"]
+
+    for friend in friends:
+        friend_movies = friend["watched"]
+        friends_all_movies +=friend_movies

```

Comment:
The instructions ask you to look at all of the movies that the user has watched that their friends have not. This sounds like what you did for `get_unique_watched`. You can call that method instead of doing this work here. On line 160 your if statement can look for movie in the unique watched list.